node/src/components/linear_chain_sync/operations.rs

```diff
@@ -588,3 +602,179 @@ where
 
     Ok(trusted_header_output)
 }
+
+/// Returns `true` if `most_recent_block` belongs to an era that is still ongoing.
+fn is_current_era<I>(
```

Comment:
I recall writing similar code - is it on this branch, too? If yes, maybe we should deduplicate it?

Reply:
It's still there, but will probably get removed before `feat-fast-sync` gets merged into `dev`. See https://github.com/casper-network/casper-node/pull/1654#discussion_r663781519.
If you also prefer the other implementation I can replace it. But I find it easier to reason about it if all the logic is in one place, and the "era height" and "era duration" criteria are separated.

Reply:
I think I actually prefer yours ;) But then I'd replace mine with yours, if it's possible. Just so that we don't have two separate implementations of the same thing.